ambry-api/src/main/java/com.github.ambry/rest/RestUtils.java

```diff
@@ -768,4 +770,12 @@ private static ByteRange buildByteRange(String rangeHeaderValue) throws RestServ
     }
     return range;
   }
+
+  /**
+   * A hacky way to determine if V3 blobId is encrypted or not.
+   */
+  public static boolean isEncrypted(String blobIdStr) {
+    return blobIdStr != null && (blobIdStr.startsWith(BLOB_ID_V3_NAIVE_ENCRYPTED_PREFIX) || blobIdStr.startsWith(
```

Comment:
This method is used together with `blobId.isEncrypted()`, for example, `blobId.isEncrypted() || RestUtils.isEncrypted(blobIdStr)`. Non-v3 is decided by blobId.isEncrypted(). 

Adding a method to BlobId.java might be OK. Open for discussion. @vgkholla 
`isEncryptedInPreamble()` ?

Construct `BlobIdPreamble` is a little time consuming since we only need a encrypted bit. Plus, `BlobIdPreamble` is a private static class now.

Reply:
Anything to do with `BlobIdPreamble` may not work. Advised Ze to put this function here since the major user (frontend) is here. Since it is also used in the router, ok if it is a static method in `BlobId`

Reply:
How do you think of adding `isEncryptedInPreamble()`(non-static) and a boolean `isEncryptedInPreamble`  to BlobId? 

Benefit is we can know the bit by `isEncryptedInPreamble()`. However, this also brings confusion between `isEncryptedInPreamble()` and `isEncrypted()`.

Reply:
Since this is a hack, i'd like it if this is all isolated to one function and is contained.

Reply:
Would strongly prefer if the whole things is exposed via a static method in `BlobId` where no other piece of code has to deal with `BlobId` versions. If the `BlobIdPreamble` has to interpret encryption as always false for V3, then please deserialize separately not via the preamble - but let us keep everything internal to `BlobId` if possible.

Reply:
So
1. V3 isEncryption parsing in `BlobIdPreamble` should be false only.
2. Move `RestUtil.isEncrypt` to BlobId. Make it a static method. May be `BlobId.isEncryptedBasedOnBlobString(String)`?
3. Should we continue use `blobId.isEncrypted() || BlobId.isEncryptedBasedOnBlobString(blobIdStr)` ? 
4. I will try parse instead of hard code 'AAME' and `AAMF` in BlobId.

Reply:
If you do something along the lines of 
`BlobId.isEncrypted(String originalBlobIdStr, BlobId constructedBlobId)`, then you can collapse #3 into this function. To prevent confusion, you can even make the `isEncrypted()` function in `BlobId` private so that this utility function is used everywhere.

Even if some piece of code does not have the "originalBlobIdStr`, it can always send in the result of `getID()`

Reply:
Thanks for `BlobId.isEncrypted(String originalBlobIdStr, BlobId constructedBlobId)`! Current `isEncrypted()` would be private.

`getID()` is calling `toBytes()`, which doesn't include encryption bit.

Reply:
right, so that is the fallback for pieces of code don't have access to the original blob id string anyway so they make do with what they have.

Reply:
Got it. Thanks!